service/src/main/java/bio/terra/profile/service/crl/GcpCrlService.java

```diff
@@ -0,0 +1,36 @@
+package bio.terra.profile.service.crl;
+
+import bio.terra.cloudres.common.ClientConfig;
+import bio.terra.cloudres.google.billing.CloudBillingClientCow;
+import bio.terra.common.iam.AuthenticatedUserRequest;
+import bio.terra.profile.service.crl.exception.CrlInternalException;
+import com.google.auth.oauth2.AccessToken;
+import com.google.auth.oauth2.GoogleCredentials;
+import java.io.IOException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+/** Provides GCP Cloud resources for the application */
+@Component
+public class GcpCrlService {
+
+  private final ClientConfig clientConfig;
+
+  @Autowired
+  public GcpCrlService(ClientConfig clientConfig) {
```

Comment:
Can you point me to code that instantiates this? I'm having trouble understanding where that is.

Reply:
It gets created automatically by spring.
This is possible because it's annotated with `@Component`, it has a constructor annotated with `@Autowired`, and there are beans set up for all of the constructor parameters.

Reply:
Ah right, magic!